### PR TITLE
Add oauth 2 Angular

### PR DESCRIPTION
**Pull request by Bolo89**
- fix https://github.com/jhipster/jhipster-lite/issues/1529

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
Can you update this PR, using the new https://github.com/jhipster/jhipster-lite/blob/main/src/main/java/tech/jhipster/lite/generator/project/domain/ProjectFile.java plz?

File: src/main/java/tech/jhipster/lite/generator/client/angular/security/oauth2/domain/AngularOauth2DomainService.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    addLoginComponentInAppHtmlFile(project);
+    addTestInAppSpecTsFile(project);
+  }
+
+  private void updateAppModuleTsFile(Project project) {
+    angularCommonService.addImports(
+      project,
+      APP_MODULE_TS_FILE_PATH,
+      """
+        import {HTTP_INTERCEPTORS} from '@angular/common/http';
```
Comment (qmonmert) [inline src/main/java/tech/jhipster/lite/generator/client/angular/security/oauth2/domain/AngularOauth2DomainService.java:86]: all imports have `space`
```diff
+ import { HTTP_INTERCEPTORS } from '@angular/common/http';
- import {HTTP_INTERCEPTORS} from '@angular/common/http';
```

File: src/main/java/tech/jhipster/lite/generator/client/angular/security/oauth2/domain/AngularOauth2DomainService.java [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        import {LoginComponent} from './login/login.component';
```
Comment (qmonmert) [inline src/main/java/tech/jhipster/lite/generator/client/angular/security/oauth2/domain/AngularOauth2DomainService.java:87]: same

File: src/main/java/tech/jhipster/lite/generator/client/angular/security/oauth2/domain/AngularOauth2DomainService.java [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        import {Oauth2AuthService} from './auth/oauth2-auth.service';
```
Comment (qmonmert) [inline src/main/java/tech/jhipster/lite/generator/client/angular/security/oauth2/domain/AngularOauth2DomainService.java:88]: same

File: src/main/java/tech/jhipster/lite/generator/client/angular/security/oauth2/domain/AngularOauth2DomainService.java [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+        import {HttpAuthInterceptor} from './auth/http-auth.interceptor';
```
Comment (qmonmert) [inline src/main/java/tech/jhipster/lite/generator/client/angular/security/oauth2/domain/AngularOauth2DomainService.java:89]: same

File: src/main/java/tech/jhipster/lite/generator/client/angular/security/oauth2/domain/AngularOauth2DomainService.java [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+    angularCommonService.addTest(
+      project,
+      APP_COMPONENT_SPEC_TS_FILE_PATH,
+      """
+          it('should display login component',() => {
+            // WHEN
+            fixture.detectChanges();
+
+            // THEN
+            expect(fixture.debugElement.query(By.directive(LoginComponent))).toBeTruthy();
```
Comment (qmonmert) [inline src/main/java/tech/jhipster/lite/generator/client/angular/security/oauth2/domain/AngularOauth2DomainService.java:170]: remove this line

File: src/main/resources/generator/client/angular/core/src/main/webapp/app/app.component.spec.ts.mustache [hunk 6]
```diff
@@ -1,4 +1,7 @@
-import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
+import {ComponentFixture, TestBed, waitForAsync} from '@angular/core/testing';
```
Comment (qmonmert) [inline src/main/resources/generator/client/angular/core/src/main/webapp/app/app.component.spec.ts.mustache:1]: same

File: src/main/resources/generator/client/angular/security/oauth2/src/main/webapp/app/auth/http-auth.interceptor.spec.ts [hunk 7]
```diff
@@ -0,0 +1,74 @@
+import { TestBed } from '@angular/core/testing';
+
+import { HttpAuthInterceptor } from './http-auth.interceptor';
+import { Oauth2AuthService } from './oauth2-auth.service';
+import { HttpHandler, HttpRequest } from '@angular/common/http';
```
Comment (qmonmert) [inline src/main/resources/generator/client/angular/security/oauth2/src/main/webapp/app/auth/http-auth.interceptor.spec.ts:5]: move with other angular imports on top

File: src/main/java/tech/jhipster/lite/generator/client/angular/security/oauth2/domain/AngularOauth2DomainService.java (hunk 5 above)
Comment (Bolo89) [reply src/main/java/tech/jhipster/lite/generator/client/angular/security/oauth2/domain/AngularOauth2DomainService.java:170]: why ?

Comment (qmonmert) [reply src/main/java/tech/jhipster/lite/generator/client/angular/security/oauth2/domain/AngularOauth2DomainService.java:170]: sorry, forget this comment

File: src/main/java/tech/jhipster/lite/generator/client/angular/common/domain/AngularCommon.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+  protected static final String TEST_REGEX_FORMAT = "([\\s]+it\\('%s',[^}]+}\\);)";
+
+  protected static final List<String> ALLOWED_COMMON_JS_DEPENDENCIES_REGEX_LIST = List.of(
+    "^([\\s]+\"allowedCommonJsDependencies\": [^\\]]+)\\]"
+  );
+
+  protected static final String COMA = ",";
+  protected static final String O_BRACKET = "[";
+  protected static final String C_BRACKET = "]";
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/client/angular/common/domain/AngularCommon.java:38]: wonder if we should move it to WordUtils ?

File: src/main/java/tech/jhipster/lite/generator/client/angular/security/oauth2/domain/AngularOauth2.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+  private AngularOauth2() {
+    // Cannot be instantiated
+  }
+
+  protected static final String APP_MODULE_TS_FILE_PATH = "src/main/webapp/app/app.module.ts";
+  protected static final String ENVIRONMENT_TS_FILE_PATH = "src/main/webapp/environments/environment.ts";
+  protected static final String ENVIRONMENT_PROD_TS_FILE_PATH = "src/main/webapp/environments/environment.prod.ts";
+  protected static final String APP_COMPONENT_HTML_FILE_PATH = "src/main/webapp/app/app.component.html";
+  protected static final String APP_COMPONENT_SPEC_TS_FILE_PATH = "src/main/webapp/app/app.component.spec.ts";
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/client/angular/security/oauth2/domain/AngularOauth2.java:16]: can use Constants.MAIN_WEBAPP

File: src/main/java/tech/jhipster/lite/generator/client/angular/security/oauth2/infrastructure/primary/AngularOauth2Resource.java [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+@Tag(name = "Angular")
+class AngularOauth2Resource {
+
+  private final AngularOauth2ApplicationService angularOauth2ApplicationService;
+
+  public AngularOauth2Resource(AngularOauth2ApplicationService angularOauth2ApplicationService) {
+    this.angularOauth2ApplicationService = angularOauth2ApplicationService;
+  }
+
+  @Operation(summary = "Add Oauth2 authentication")
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/client/angular/security/oauth2/infrastructure/primary/AngularOauth2Resource.java:27]:
```suggestion
  @Operation(summary = "Add OAuth2 authentication")
```

File: src/main/java/tech/jhipster/lite/generator/client/angular/security/oauth2/infrastructure/primary/AngularOauth2Resource.java [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+  @ApiResponse(responseCode = "500", description = "An error occurred while adding Oauth2 authentication")
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/client/angular/security/oauth2/infrastructure/primary/AngularOauth2Resource.java:28]:
```suggestion
  @ApiResponse(responseCode = "500", description = "An error occurred while adding OAuth2 authentication")
```

File: src/main/webapp/app/springboot/primary/generator/angular-generator/AngularGenerator.vue [hunk 12]
```diff
@@ -2,11 +2,12 @@
   <div id="v-pills-angular" class="tab-pane fade" role="tabpanel" aria-labelledby="v-pills-settings-tab">
     <GeneratorButtonVue :label="'Generate Angular'" :service="'angular'" :selector-prefix="selectorPrefix" @click.prevent="addAngular" />
     <GeneratorButtonVue
-      :label="'Add Angular with JWT'"
+      :label="'Add JWT'"
       :service="'angular-with-jwt'"
       :selector-prefix="selectorPrefix"
       @click.prevent="addAngularWithJWT"
     />
+    <GeneratorButtonVue :label="'Add Oauth2'" :service="'angular-oauth2'" :selector-prefix="selectorPrefix" @click.prevent="addOauth2" />
```
Comment (pascalgrimaud) [inline src/main/webapp/app/springboot/primary/generator/angular-generator/AngularGenerator.vue:10]:
```suggestion
    <GeneratorButtonVue :label="'Add OAuth2'" :service="'angular-oauth2'" :selector-prefix="selectorPrefix" @click.prevent="addOauth2" />
```

File: src/test/java/tech/jhipster/lite/generator/client/angular/common/application/AngularCommonApplicationServiceIT.java [hunk 13]
```diff
@@ -0,0 +1,285 @@
...
+import tech.jhipster.lite.generator.init.application.InitApplicationService;
+import tech.jhipster.lite.generator.project.domain.Project;
+
+@IntegrationTest
+class AngularCommonApplicationServiceIT {
+
+  private static final String APP_MODULE_TS_FILE_PATH = "src/main/webapp/app/app.module.ts";
+  private static final String ENVIRONMENT_TS_FILE_PATH = "src/main/webapp/environments/environment.ts";
+  private static final String APP_COMPONENT_HTML_FILE_PATH = "src/main/webapp/app/app.component.html";
+  private static final String APP_COMPONENT_SPEC_TS_FILE_PATH = "src/main/webapp/app/app.component.spec.ts";
```
Comment (pascalgrimaud) [inline src/test/java/tech/jhipster/lite/generator/client/angular/common/application/AngularCommonApplicationServiceIT.java:25]: can use Constants.MAIN_WEBAPP

File: src/test/java/tech/jhipster/lite/generator/client/angular/common/application/AngularCommonApplicationServiceIT.java [hunk 14]
```diff
@@ -0,0 +1,285 @@
...
+  private static final String ANGULAR_JSON_FILE_PATH = "angular.json";
```
Comment (pascalgrimaud) [inline src/test/java/tech/jhipster/lite/generator/client/angular/common/application/AngularCommonApplicationServiceIT.java:26]: no need this, as there is already ANGULAR_JSON_FILE_NAME ?

File: src/test/java/tech/jhipster/lite/generator/client/angular/security/oauth2/application/AngularOauth2Assert.java [hunk 15]
```diff
@@ -0,0 +1,155 @@
...
+public class AngularOauth2Assert {
+
+  private static final List<String> ADDED_FILES_PATHS = List.of(
+    "src/main/webapp/app/auth/oauth2-auth.service.ts",
+    "src/main/webapp/app/auth/oauth2-auth.service.spec.ts",
+    "src/main/webapp/app/auth/http-auth.interceptor.ts",
+    "src/main/webapp/app/auth/http-auth.interceptor.spec.ts",
+    "src/main/webapp/app/login/login.component.html",
+    "src/main/webapp/app/login/login.component.ts",
+    "src/main/webapp/app/login/login.component.spec.ts"
```
Comment (pascalgrimaud) [inline src/test/java/tech/jhipster/lite/generator/client/angular/security/oauth2/application/AngularOauth2Assert.java:25]: use Constants.MAIN_WEBAPP

File: src/test/javascript/cypress/integration/Generator.spec.ts [hunk 16]
```diff
@@ -60,7 +60,8 @@ describe('Generator', () => {
   it('should display angular', () => {
     cy.get(generatorSelector('option-angular')).check();
     cy.get(angularGeneratorSelector('add-angular-button')).contains('Angular');
-    cy.get(angularGeneratorSelector('add-angular-with-jwt-button')).contains('Angular with JWT');
+    cy.get(angularGeneratorSelector('add-angular-with-jwt-button')).contains('Add JWT');
+    cy.get(angularGeneratorSelector('add-angular-oauth2-button')).contains('Add Oauth2');
```
Comment (pascalgrimaud) [inline src/test/javascript/cypress/integration/Generator.spec.ts:64]:
```suggestion
    cy.get(angularGeneratorSelector('add-angular-oauth2-button')).contains('Add OAuth2');
```

**Review by pascalgrimaud — APPROVED** [review]

**Comment by pascalgrimaud** [comment]
move the bounty to the ticket, not to this PR :D

**Comment by pascalgrimaud** [comment]
thanks a lot for your work @Bolo89 
it's awesome!

**Comment by Bolo89** [comment]
Bounty claimed at https://opencollective.com/generator-jhipster/expenses/84925

**Comment by pascalgrimaud** [comment]
@Bolo89 : approved, thanks for your work

**Pull request merged**